### PR TITLE
Add forum appearance restoration option

### PR DESCRIPTION
**Pull request by thekifake**
This adds an option in configuration to restore the way the forum homepage looked before it got modified. In the event that the forum thread pages are archived it might be necessary to take further steps to restore the look as it takes its appearance from the /things/ page. I have the current stylesheets from November 2022 archived, though, so if this does happen it shouldn't be that big of a deal.

### CODE REVIEW

**Comment by aprzn123** [comment]
Something feels off to me - did the forum home page not have this same dark-gray bar at the top?
(first image: forum home; second image: subforum)

image

image

**Comment by thekifake** [comment]
No, it didn't. I had noiwillnot send me a screenshot of how it looked before the change and the top bar was not black. If you feel the need to remove the inline styles to make it black again for consistency though then feel free to do so.

**Comment by aprzn123** [comment]
Can you send that screenshot? Because something still feels off but I can't quite put my finger on it

**Comment by aprzn123** [comment]
Oh I think the issue is that the number columns should be left-aligned but you have them centered. Can you try changing that and see if it fixes it?

File: src/resource/fourth-api.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+ * @example fourth.Request("forum.categories");
+ */
+fourth.Request = async function(fn) {
+  if (typeof fn === "string") fn = [fn];
+  let fns = [];
+  for (let f of fn) {
+    fns.push({"fn": f});
+  }
+  return new Promise((resolve, reject) => {
+    const xhr = new XMLHttpRequest();
```
Comment (aprzn123) [inline src/resource/fourth-api.js:17]: what is the motivation for using XHR over fetch?

**Comment by HumanoidSandvichDispenser** [comment]
Could you cherry pick commits for the forum appearance into one branch and fourth API to another?

**Review by aprzn123 — COMMENTED** [review]
honestly this is really good, I'm impressed. I just have the one minor question about the fourth API implementation, but other than that it's pretty great!

**Comment by aprzn123** [comment]
> Could you cherry pick commits for the forum appearance into one branch and fourth API to another? It will be easier to review and merge each one separately and resolve merge conflicts on local repositories in case there will be one. Thanks.

I can manage fine, it's a bit larger scope than might be preferable but this is a small project anway

File: src/resource/fourth-api.js (hunk 1 above)
Comment (aprzn123) [reply src/resource/fourth-api.js:17]: It's not strictly necessary, but I do usually prefer fetch. maybe a quick switch over, and then merge?

Comment (thekifake) [reply src/resource/fourth-api.js:17]: Alright, sure.

**Comment by aprzn123** [comment]
One more thing--I'm not an expert with CSS, but is there an easy way to get this to wrap on the larger scale instead of on each individual username?
image

**Comment by thekifake** [comment]
Check now.

**Review by aprzn123 — APPROVED** [review]
LGTM!

**Pull request merged**